### ISSUE TITLE
_.deferred for wrapping slow functions

### ISSUE THREAD

**Issue by lemonzi** [jashkenas/underscore#1212]
Let me show directly the use case:

Event-driven programming makes us think in parallel, but the callbacks are usually executed one after another and synchronously. That means the system will be blocked until all of them have been executed. 

I suggest adding `_.deferred(function)`, which returns a function that always waits until the next iteration of the event loop before executing its argument. Then, if we have slow callbacks, avoiding blocking is only a matter of making them deferred when binding them to the event.

So, it would generate a function that, when called, would be like calling `_.defer()` on the original function.

Maybe it's too specific, or too easy to implement using `_.defer`... What do you think?
